File: src/leetcode/2026/04/Solution796.cpp
#include <string>
using namespace std;

class Solution796 {
public:
    bool rotateString(string s, string goal) {
        if (s.length() != goal.length()) return false;
        string doubled = s + s;
        return doubled.find(goal) != string::npos;
    }
};

int main() {
    Solution796 solution;

    string s = "abcde";
    string goal = "cdeab";
    bool result = solution.rotateString(s, goal);
    printf("Can rotate string: %s\n", result ? "true" : "false"); // Output: true

    goal = "abced";
    result = solution.rotateString(s, goal);
    printf("Can rotate string: %s\n", result ? "true" : "false"); // Output: false

    return 0;
}

File: src/leetcode/2026/04/Solution796.java
public class Solution796 {
    public boolean rotateString(String s, String goal) {
        if (s.length() != goal.length()) {
            return false;
        }

        String concatenated = s + s;
        return concatenated.contains(goal);
    }

    public static void main(String[] args) {
        Solution796 solution = new Solution796();

        String s1 = "abcde";
        String goal1 = "cdeab";
        System.out.println(solution.rotateString(s1, goal1)); // Output: true

        String s2 = "abcde";
        String goal2 = "abced";
        System.out.println(solution.rotateString(s2, goal2)); // Output: false
    }
}
